scheduler/scheduler-examples/src/main/java/org/ow2/proactive/scheduler/examples/FTPConnector.java

```diff
@@ -0,0 +1,374 @@
+                                }
+
+                                Files.copy(zipFile.getInputStream(entry), path);
+                            }
+                        }
+                    } catch (IOException ex) {
+                        ex.printStackTrace();
+                    } finally {
+                        zipFile.close();
+
+                    }
+                }
+            } // If it is a folder, download all its contents recursively
+            else {
+                ftpClient.changeWorkingDirectory(ftpRemoteFilePath);
+                downloadDirectory(ftpClient, "", "", ftpLocalFilePath);
+            }
+            getOut().println("END Import file(s) from FTP.");
+        }
+        //FTP mode is PUT
+        else if (ftpMode.equals("PUT")) {
+            if (ftpLocalFilePath.isEmpty()) {
+                throw new IllegalArgumentException("The ftp local file path variable is empty. Please, enter a value.");
+            }
+            getOut().println("Exporting file(s) from " + ftpLocalFilePath + " to " + ftpRemoteFilePath);
+            File localFile = new File(ftpLocalFilePath);
+            // if the ftp remote file path does not exist then it is created.
+            if (!checkDirectoryExists(ftpClient, ftpRemoteFilePath)) {
+                ftpClient.makeDirectory(ftpRemoteFilePath);
+                getOut().println(ftpRemoteFilePath + " does not exist. Therefore, it is created.");
+            } else {
+                getOut().println(ftpRemoteFilePath + " exists.");
+            }
+            // If it is a single file:
+            if (localFile.isFile()) {
+                getOut().println("A single FILE to upload");
+                ftpClient.changeWorkingDirectory(ftpRemoteFilePath);
+                boolean uploaded = uploadSingleFile(ftpClient, ftpLocalFilePath, localFile.getName());
+                if (uploaded) {
+                    getOut().println("UPLOADED a file to: " + Paths.get(ftpRemoteFilePath, localFile.getName()));
+                } else {
+                    getOut().println("COULD NOT upload the file: " + ftpLocalFilePath);
+                }
+            }
+            // If it is a folder, upload all its contents recursively
+            else if (localFile.isDirectory()) {
+                getOut().println("A DIRECTORY to upload");
+                ftpClient.changeWorkingDirectory(ftpRemoteFilePath);
+                uploadDirectory(ftpClient, "", ftpLocalFilePath, "");
+            } else {
+                throw new IllegalArgumentException(localFile + " not found. Please, enter a valid path.");
+            }
+            getOut().println("END Export file(s) to FTP.");
+        }
+        // log out and disconnect from the server
+        ftpClient.logout();
+        ftpClient.disconnect();
+
+        getOut().println("Disconnected");
+
+        return 1;
+    }
+
+    private void uploadDirectory(FTPClient ftpClient, String remoteDirPath, String localParentDir,
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this method to reduce its Cognitive Complexity from 25 to the 15 allowed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3776)